installation/proc-add-custom-app-config-file-ocp.adoc

```diff
@@ -0,0 +1,86 @@
+
+* Add a custom application configuration file to {ocp-short} and reference it in the Custom Resource (CR). In {ocp-short}, you can use the following example as a base template to create a ConfigMap such as `app-config-rhdh.yaml`:
++
+[source,yaml,subs="attributes+"]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: app-config-rhdh
+data:
+  "app-config-rhdh.yaml": |
+    app:
+      title: {product}
+      baseUrl: https://backstage-developer-hub-my-ns.apps.ci-ln-vtkzr22-72292.origin-ci-int-gce.dev.rhcloud.com
+    backend:
+      auth:
+        keys:
+          - secret: "${BACKEND_SECRET}"
+      baseUrl: https://backstage-backstage-sample-my-ns.apps.ci-ln-vtkzr22-72292.origin-ci-int-gce.dev.rhcloud.com
+      cors:
+        origin: https://backstage-backstage-sample-my-ns.apps.ci-ln-vtkzr22-72292.origin-ci-int-gce.dev.rhcloud.com
+----
++
+* Use the mandatory backend auth key for {product} to reference an environment variable defined in an {ocp-short} Secret.
+* Set the external URL of your {product} instance in the `app.baseUrl`, `backend.baseUrl` and `backend.cors.origin` fields of the application configuration. By default, the URL is similar to the following example:
+`pass:c[https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_INGRESS_DOMAIN>;]`.
+** You can use the `oc get ingresses.config/cluster -o jsonpath='{.spec.domain}'` command to display your ingress domain. If you want to use a different host or sub-domain, customize the `Custom Resource spec.application.route field` and adjust the application configuration accordingly.
+
+[NOTE]
+You are responsible for protecting your {product} installation from external and unauthorized access. Manage the backend auth key like any other secret. Meet strong password requirements, do not expose it in any configuration files, and only inject it into configuration files as an environment variable.
+
+.Prerequisites
+* You have an active {ocp-brand-name} account.
+
+.Procedure
+. From the *Developer* perspective, select the *ConfigMaps* tab.
+. Click *Create ConfigMap*.
+. Select the *YAML view* option in *Configure via* and make the changes to the file, if necessary.
+. Click *Create*.
+. Select the *Secrets* tab.
+. Click *Create Key/value Secret*.
+. Name the secret `secrets-rhdh`.
+. Add a key named `BACKEND_SECRET` and a base64 encoded string as a value. Use a unique value for each {product} instance. For example, you can use the following command to generate a key from your terminal:
++
+[source,yaml]
+----
+node -p 'require("crypto").randomBytes(24).toString("base64")'
+----
+
+. Click *Create*.
+. Select the *Topology* tab.
+. Click the overflow menu for the {product} instance that you want to use and select *Edit Backstage* to load the YAML view of the {product} instance.
++
+image::rhdh/operator-install-2.png[]
+
+. Add the `spec.application.appConfig.configMaps` and `spec.application.extraEnvs.secrets` fields to the Custom Resource. For example:
```

Comment:
```suggestion
. Add the `spec.application.appConfig.configMaps` and `spec.application.extraEnvs.secrets` fields to the custom resource. For example:
```

See https://redhat-documentation.github.io/supplementary-style-guide/#custom-resource for guidance about using the term "custom resource". Generally it's lower case, and for the first instance in a module/assembly, we write it out as "custom resource (CR)" and then just refer to it as "CR" for the rest of the doc.